packages/api/src/command/medical/patient/upsert-geographic-coordinates.ts

```diff
@@ -63,19 +66,21 @@ export async function addCoordinatesToAddresses({
   patient: Partial<Patient>;
   reportRelevance?: boolean;
 }): Promise<Address[]> {
-  const addressAndLabel = await addGeographicCoordinates(addresses);
-  if (reportRelevance) await reportLowRelevance(addressAndLabel, patient);
-  return addresses;
+  const geocodedAddressResults = await generateGeocodedAddresses(addresses);
+  if (reportRelevance) await reportLowRelevance(geocodedAddressResults, patient);
+
+  const updatedAddresses = geocodedAddressResults.map(p => p.address);
+  const addressesWithCoordinates = uniqBy([...updatedAddresses, ...addresses], "addressLine1");
```

Comment:
It might be kinda difficult to explain in words, but I will try...

So patient's addresses might or might not have coordinates. The `updatedAddresses` in this function contains a list of only those addresses that got geocoded during this function invocation. Thus, if you combine original addresses with updated addresses, you will have a list of all of the original addresses + duplicates with coordinates for the ones that were successfully geocoded. 

Thus, we need to dedup those addresses and make sure we're keeping the ones that have coordinates, while booting the ones that don't. 

However, that might also mean that if a CX-entered address fails to receive coordinates for whatever reason, it will be filtered out and erased from the patient's record. Thus, we need to filter out all duplicate addresses without coordinates, while keeping unique addresses even if they don't have coordinates. 

Hope this makes sense... Happy to chat if not 😃 

I updated the function correspondingly, but tbh not too happy with how it looks myself..

Reply:
![addresses](https://github.com/metriport/metriport/assets/90670087/b75db65d-ed3e-45f8-8571-67e8471d4580)

Reply:
That makes sense, thanks for the explanation!

Reply:
One thing that I realized is: we don't have the concept of main/principal/current address, and we use a few `address[0]` throughout the codebase (hopefully going to be updated soon).

This means cxs might chose to send a patient's address in order of importance. And this code changes that.

We could achieve the same result by doing it slightly differently:
- create a resulting list
- loop through the original address list
- for each, see if there's a matching address on the list of geocoded ones
- if yes, add the geocoded one to the result list
- if not, add the current address from the original list